examples/md-flexible/src/Simulation.cpp

```diff
@@ -21,525 +22,189 @@ extern template bool autopas::AutoPas<ParticleType>::iteratePairwise(autopas::LJ
-  cout << timerToString("    Force       ", _timers.forceUpdateTotal.getTotalTime(), digitsTimeTotalNS,
-                        durationSimulate);
-  cout << timerToString("      Tuning    ", _timers.forceUpdateTuning.getTotalTime(), digitsTimeTotalNS,
-                        _timers.forceUpdateTotal.getTotalTime());
-  cout << timerToString("      NonTuning ", _timers.forceUpdateNonTuning.getTotalTime(), digitsTimeTotalNS,
-                        _timers.forceUpdateTotal.getTotalTime());
-  cout << timerToString("    Velocity    ", _timers.velocityUpdate.getTotalTime(), digitsTimeTotalNS, durationSimulate);
-  cout << timerToString("    VTK         ", _timers.vtk.getTotalTime(), digitsTimeTotalNS, durationSimulate);
-  cout << timerToString("    Thermostat  ", _timers.thermostat.getTotalTime(), digitsTimeTotalNS, durationSimulate);
-
-  cout << timerToString("One iteration   ", _timers.simulate.getTotalTime() / _iteration, digitsTimeTotalNS,
-                        durationTotal);
-  auto mfups = autopas.getNumberOfParticles(autopas::IteratorBehavior::owned) * _iteration * 1e-6 /
-               (_timers.forceUpdateTotal.getTotalTime() * 1e-9);  // 1e-9 for ns to s, 1e-6 for M in MFUP
-  cout << "Tuning iterations: " << _numTuningIterations << " / " << _iteration << " = "
-       << ((double)_numTuningIterations / _iteration * 100) << "%" << endl;
-  cout << "MFUPs/sec    : " << mfups << endl;
-
-  if (_config->dontMeasureFlops.value) {
-    autopas::FlopCounterFunctor<ParticleType> flopCounterFunctor(autopas.getCutoff());
-    autopas.iteratePairwise(&flopCounterFunctor);
-
-    auto flops = flopCounterFunctor.getFlops(flopsPerKernelCall) * _iteration;
-    // approximation for flops of verlet list generation
-    if (autopas.getContainerType() == autopas::ContainerOption::verletLists)
-      flops += flopCounterFunctor.getDistanceCalculations() *
-               decltype(flopCounterFunctor)::numFlopsPerDistanceCalculation *
-               floor(_iteration / _config->verletRebuildFrequency.value);
-
-    cout << "GFLOPs       : " << flops * 1e-9 << endl;
-    cout << "GFLOPs/sec   : " << flops * 1e-9 / durationSimulateSec << endl;
-    cout << "Hit rate     : " << flopCounterFunctor.getHitRate() << endl;
-  }
+  _timers.initialization.stop();
 }
 
-const std::unique_ptr<ParticlePropertiesLibrary<double, size_t>> &Simulation::getPpl() const {
-  return _particlePropertiesLibrary;
-}
+Simulation::~Simulation() { _timers.total.stop(); }
 
-std::string Simulation::timerToString(const std::string &name, long timeNS, size_t numberWidth, long maxTime) {
-  // only print timers that were actually used
-  if (timeNS == 0) {
-    return "";
+void Simulation::run() {
+  const int iterationsPerSuperstep = _configuration.verletRebuildFrequency.value;
+  _timers.simulate.start();
+  for (int i = 0; i < _configuration.iterations.value; i += iterationsPerSuperstep) {
```

Comment:
I just noticed that you basically changed the meaning of configuration values. "iterations" is actually intended to be the number of iterations of the simulation. So if I do `md-flexible --iterations 2` I expect the simulation to do 2 iterations.

Reply:
```suggestion
  while(needsMoreIterations()) {
```
That should do it. I still think this "superstep" loop is weird. Any reason not to merge it with the loop from line 158?

Reply:
I will remove  the executeSuperStep function and implement your suggestion. I am not sure anymore what I had in mind when implementing it.